test/xbgpu/test_engine.py

```diff
@@ -638,8 +668,23 @@ async def _send_data(
 
                 corrprod_results[i][j] = ig_recv["xeng_raw"].value
 
+        # TODO: NGC-1172 The tweaks to process beam data below rely on
+        # `batch_indices` to be contiguous. The check below is temporary until
```

Comment:
If I'm not mistaken, it also depends on `batch_indices[0]` being aligned to a multiple of HEAPS_PER_FENGINE_PER_CHUNK.

Reply:
Too right, although I would say that's nothing overly concerning
* We've just done that as it's a convenient (reliable/repeatable) way of forcing a test of the XPipeline (Re: that old ~resync logic comment).

I'm not sure it requires a tweak to the comment here
* It's a relatively unrelated point that shouldn't cause an issue for this check (?)
* But perhaps worth a mention, regardless